internal/engine/compiler/engine.go

```diff
@@ -808,7 +817,7 @@ func newEngine(ctx context.Context, enabledFeatures api.CoreFeatures) *engine {
 		enabledFeatures: enabledFeatures,
 		codes:           map[wasm.ModuleID][]*code{},
 		setFinalizer:    runtime.SetFinalizer,
-		Cache:           compilationcache.NewFileCache(ctx),
+		Cache:           fileCache,
```

Comment:
unless fileCache will only be for files, then rename field to FileCompileCache

```suggestion
		CompileCache:           fileCache,
```